pennylane/operation.py

```diff
@@ -300,48 +300,149 @@ def hash(self):
+                [ 0.24740396,  0.96891242, 0., 0.],
+                [ 0., 0., 0.96891242, -0.24740396],
+                [ 0., 0., 0.24740396,  0.96891242]
+                ])
+
+        >>> qml.RY.compute_matrix(0.5, wires=["b"], wire_order=["b", "a"])
+        tensor([[ 0.96891242, 0., -0.24740396, 0.],
+                [0.,  0.96891242, 0., -0.24740396],
+                [ 0.24740396, 0., 0.96891242, 0.],
+                [0.,  0.24740396, 0., 0.96891242+0.j],
+                ])
+
+        Args:
+            params (Iterable): trainable parameters that influence the matrix representation
+            wires (Iterable): wires of this operator
+            wire_order (Iterable): global wire order, must contain all wire labels in ``wires``
+            hyperparams (dict): other inputs that influence the matrix representation
 
         Returns:
-            array: matrix representation
+            tensor-like: matrix representation, or None if not defined
         """
-        raise NotImplementedError
+        base_matrix = cls._matrix(*params, **hyperparams)
+        interface = qml.math._multi_dispatch(params)  # pylint: disable=protected-access
 
-    @property
-    def matrix(self):
+        # representation is not defined
+        if base_matrix is None:
+            return None
+
+        # cannot have wire_order but no wires to interpret
+        if wires is None and wire_order is not None:
+            raise ValueError("Expected wires when wire order is defined, got None.")
+
+        # intersecting this case here because it is common
+        # and we want to avoid casting to Wires unless necessary
+        if wire_order is None:
+            # TODO: cast to interface?
+            return base_matrix
+
+        wire_order = Wires(wire_order)
+        wires = Wires(wires)
+
+        # base matrix requested
+        if Wires(wires) == Wires(wire_order):
+            # TODO: cast to interface?
+            return base_matrix
+
+        n = len(wires)
+
+        # operator's wire positions relative to wire ordering
+        op_wire_pos = Wires(wire_order).indices(wires)
+
+        I = qml.math.reshape(
+            qml.math.eye(2 ** len(wire_order), like=interface), [2] * len(wire_order) * 2
+        )
+        axes = (list(range(n, 2 * n)), op_wire_pos)
+
+        # reshape op.matrix()
+        op_matrix_interface = qml.math.convert_like(base_matrix, I)
+        mat_op_reshaped = qml.math.reshape(op_matrix_interface, [2] * n * 2)
+        mat_tensordot = qml.math.tensordot(
+            mat_op_reshaped, qml.math.cast_like(I, mat_op_reshaped), axes
+        )
+
+        unused_idxs = [idx for idx in range(len(wire_order)) if idx not in op_wire_pos]
+        # permute matrix axes to match wire ordering
+        perm = op_wire_pos + unused_idxs
+        mat = qml.math.moveaxis(mat_tensordot, wire_order.indices(wire_order), perm)
+
+        mat = qml.math.reshape(mat, (2 ** len(wire_order), 2 ** len(wire_order)))
```

Comment:
@mariaschuld I wonder if either of 

https://github.com/PennyLaneAI/pennylane/blob/1fd370d0ee2b14c3ddef9af6e5eccbf8784350a6/pennylane/utils.py#L307-L308

or 

https://github.com/PennyLaneAI/pennylane/blob/1fd370d0ee2b14c3ddef9af6e5eccbf8784350a6/pennylane/utils.py#L372-L373

can be used here?

Although, your implementation might be preferred since it uses `qml.math` :)

Reply:
I copied it from Olivia's code to get a unitary. Will outsource it!